// @nestjs/core/router/route-path-factory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoutePathFactory = void 0;
const common_1 = require("@nestjs/common");
const shared_utils_1 = require("@nestjs/common/utils/shared.utils");
class RoutePathFactory {
    constructor(applicationConfig) {
        this.applicationConfig = applicationConfig;
    }
    create(metadata, requestMethod) {
        var _a;
        let paths = [''];
        const version = this.getVersion(metadata);
        if (version && ((_a = metadata.versioningOptions) === null || _a === void 0 ? void 0 : _a.type) === common_1.VersioningType.URI) {
            const versionPrefix = this.getVersionPrefix(metadata.versioningOptions);
            // Version Neutral - Do not include version in URL
            if (version !== common_1.VERSION_NEUTRAL) {
                if (Array.isArray(version)) {
                    paths = common_1.flatten(paths.map(path => version.map(v => path + `/${versionPrefix}${v}`)));
                }
                else {
                    paths = paths.map(path => path + `/${versionPrefix}${version}`);
                }
            }
        }
        paths = this.appendToAllIfDefined(paths, metadata.modulePath);
        paths = this.appendToAllIfDefined(paths, metadata.ctrlPath);
        paths = this.appendToAllIfDefined(paths, metadata.methodPath);
        if (metadata.globalPrefix) {
            paths = paths.map(path => {
                if (this.isExcludedFromGlobalPrefix(path, requestMethod)) {
                    return path;
                }
                return shared_utils_1.stripEndSlash(metadata.globalPrefix || '') + path;
            });
        }
        return paths
            .map(path => shared_utils_1.addLeadingSlash(path || '/'))
            .map(path => (path !== '/' ? shared_utils_1.stripEndSlash(path) : path));
    }
    getVersion(metadata) {
        // The version will be either the path version or the controller version,
        // with the pathVersion taking priority.
        return metadata.methodVersion || metadata.controllerVersion;
    }
    getVersionPrefix(versioningOptions) {
        const defaultPrefix = 'v';
        if (versioningOptions.type === common_1.VersioningType.URI) {
            if (versioningOptions.prefix === false) {
                return '';
            }
            else if (versioningOptions.prefix !== undefined) {
                return versioningOptions.prefix;
            }
        }
        return defaultPrefix;
    }
    appendToAllIfDefined(paths, fragmentToAppend) {
        if (!fragmentToAppend) {
            return paths;
        }
        const concatPaths = (a, b) => shared_utils_1.stripEndSlash(a) + shared_utils_1.addLeadingSlash(b);
        if (Array.isArray(fragmentToAppend)) {
            const paths2dArray = paths.map(path => fragmentToAppend.map(fragment => concatPaths(path, fragment)));
            return common_1.flatten(paths2dArray);
        }
        return paths.map(path => concatPaths(path, fragmentToAppend));
    }
    isExcludedFromGlobalPrefix(path, requestMethod) {
        if (shared_utils_1.isUndefined(requestMethod)) {
            return false;
        }
        const options = this.applicationConfig.getGlobalPrefixOptions();
        if (!options.exclude) {
            return false;
        }
        return options.exclude.some(route => {
            if (!route.pathRegex.exec(path)) {
                return false;
            }
            return (route.requestMethod === common_1.RequestMethod.ALL ||
                route.requestMethod === requestMethod);
        });
    }
}
exports.RoutePathFactory = RoutePathFactory;
