diff --git a/./rest.py b/./rest.py
index d31433eb..b9e76e60 100644
--- a/./rest.py
+++ b/./rest.py
@@ -39,11 +39,11 @@ class RESTResponse(io.IOBase):

     def getheaders(self):
         """Returns a dictionary of the response headers."""
-        return self.urllib3_response.getheaders()
+        return self.urllib3_response.headers

     def getheader(self, name, default=None):
         """Returns a given response header."""
-        return self.urllib3_response.getheader(name, default)
+        return self.urllib3_response.headers.get(name, default)

